pkgs/development/ocaml-modules/noise/default.nix

```diff
@@ -0,0 +1,59 @@
+{ lib
+, buildDunePackage
+, fetchFromGitHub
+
+, benchmark
+, callipyge
+, chacha
+, digestif
+, lwt
+, lwt_ppx
+, nocrypto
+, ounit
+, ppx_let
+, ppx_deriving_yojson
+, rfc7748
+, tweetnacl
+}:
+
+buildDunePackage rec {
+  pname = "noise";
+  version = "0.2.0";
+
+  src = fetchFromGitHub {
+    owner = "emillon";
+    repo = "ocaml-noise";
+    rev = "v${version}";
+    sha256 = "sha256-RhUtEh8sOzXIFTvDBXbeg6h+DwVJIJkilhQmYB6V/fM=";
+  };
+
+  useDune2 = true;
+
+  minimumOCamlVersion = "4.05";
+
+  doCheck = true;
+  checkInputs = [
+    benchmark
+    lwt_ppx
+    ounit
+    ppx_deriving_yojson
+  ];
+
+  propagatedBuildInputs = [
+    callipyge
+    chacha
+    digestif
+    lwt
+    nocrypto
+    ppx_let
```

Comment:
`ppx_let` is a `nativeBuildInput` most likely not specified correctly in the opam file.